src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9160,121 +9159,58 @@ namespace SurfaceGeometry {
-                                                                             state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(Vrt).y,
-                                                                             state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(Vrt).z));
-                        ShowContinueError(state,
-                                          format("Vertex [{}]=({:.2R},", Vrt - 1, state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(Vrt - 1).x) +
-                                              format("{:.2R},{:.2R})",
-                                                     state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(Vrt - 1).y,
-                                                     state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex(Vrt - 1).z));
+            auto &surface = state.dataSurfaceGeometry->SurfaceTmp(SurfNum);
+            auto &vertices = surface.Vertex;
+            auto &nSides = surface.Sides;
+
+            bool poppedVertex = true;
+            while (poppedVertex) {
+                poppedVertex = false;
+                Perimeter = 0.0;
+
+                for (auto it = vertices.begin(); it != vertices.end(); ++it) {
+                    auto itnext = std::next(it);
+                    if (itnext == std::end(vertices)) {
+                        itnext = std::begin(vertices);
                     }
-                    ++state.dataErrTracking->TotalCoincidentVertices;
-                    if (Vrt == state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides) {
-                        if (state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides > 3) {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(state, format("Dropping Vertex [{}].", state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides));
-                            }
-                            --state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides;
-                            state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Vertex.redimension(state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides);
-                        } else {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(
-                                    state,
-                                    format("Cannot Drop Vertex [{}]; Number of Surface Sides at minimum. This surface is now a degenerate surface.",
-                                           state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Sides));
-                            }
-                            ++state.dataErrTracking->TotalDegenerateSurfaces;
-                            // mark degenerate surface?
+
+                    // TODO: use isAlmostEqual3Pt for consistency? (which uses 0.0127 m / 1/2inch instead of 0.01 m)
+                    DistanceCheck = distance(*it, *itnext);
```

Comment:
Alright, so new distance check is here, this looks really nice and clearly shows that it is operating on the current and next.